kernel/include/mmu.h

```diff
@@ -98,78 +105,89 @@
  */
 struct z_page_frame {
 	union {
-		/* If mapped, virtual address this page is mapped to */
-		void *addr;
-
-		/* If unmapped and available, free pages list membership. */
-		sys_snode_t node;
+		/*
+		 * If mapped, Z_PAGE_FRAME_* flags and virtual address
+		 * this page is mapped to.
+		 */
+		uintptr_t va_and_flags;
```

Comment:
Since you are now using the lower bits as flags, with the bits above, the minimum page frame size is now 128 bytes. I don't think any arch would have page size smaller than 4096, but it might be a good idea to add a build assert... just in case.